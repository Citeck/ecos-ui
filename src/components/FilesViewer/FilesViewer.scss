@import '../../styles/constants';

.ecos-files-viewer {
  &_dragged-not-files {
    padding-bottom: 30px;
  }

  &__item {
    $item: &;

    user-select: none;
    display: grid;
    grid-auto-flow: column;
    align-items: center;
    height: 40px;
    color: $text-color-dark;
    font-size: $font-size-standard;
    font-weight: $font-weight-semi-bold;
    cursor: pointer;
    grid-template-columns: 60% auto;

    &_selected {
      background-color: $color-yellow;
    }

    &-icon {
      color: var(--light-primary-color);

      &-wrapper {
        width: 50px;
        height: 100%;
        display: flex;
        justify-content: center;
        align-items: center;
      }
    }

    &-file-icon {
      width: 1.8em;
      height: 100%;
    }

    &-left {
      display: grid;
      grid-auto-flow: column;
      align-items: center;
      justify-content: left;
      height: 100%;
    }

    &-right {
      @extend %ellipsis;
      position: relative;
      display: grid;
      align-items: center;
      justify-content: start;
      grid-auto-flow: column;
      height: 100%;

      &_mobile {
        justify-content: center;
      }
    }

    &-title {
      @extend %ellipsis;
    }

    &-actions {
      display: none;
    }

    &:hover,
    &_mobile#{$item}_lastclicked {
      outline: 1px solid $color-brown-2;
      outline-offset: -1px;

      #{$item}-actions {
        display: inline-flex;
        position: absolute;
        right: 2px;
        top: 2px;
        bottom: 2px;
        padding-left: 20px;
        background: linear-gradient(to right, rgba($color-white, 0), $color-white 15%);
      }

      &#{$item}_selected {
        #{$item}-actions {
          background: linear-gradient(to right, rgba($color-yellow-1, 0), $color-yellow-1 15%);
        }
      }
    }
  }
}
