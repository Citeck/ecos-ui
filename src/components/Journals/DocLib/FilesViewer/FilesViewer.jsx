import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import isEmpty from 'lodash/isEmpty';
import get from 'lodash/get';

import { Well } from '../../../common/form';
import Loader from '../../../common/Loader/Loader';
import { t } from '../../../../helpers/export/util';

import FileList from './FileList';
import './FileViewer.scss';
import Empty from './Empty';

const FilesViewer = ({ isMobile, fileViewer = {}, openFolder, setSelected, setLastClicked, groupActions, path, onDrop, isLoading }) => {
  const { hasError, isReady, items, selected, lastClicked } = fileViewer;

  let content;

  if (hasError) {
    content = t('document-library.failure-to-fetch-data');
  } else if (!isReady) {
    content = <Loader blur rounded style={{ position: 'relative', margin: '0.5em 0' }} />;
  } else {
    const isGroupActionsReady = get(groupActions, 'isReady', false);
    const forRecords = get(groupActions, 'forRecords', {});
    const isGroupActionsVisible = !isGroupActionsReady || !isEmpty(forRecords.actions);
    const isBreadcrumbsVisible = !!path.length;

    content =
      items.length > 0 ? (
        <div
          className={classNames('ecos-doclib__fileviewer', {
            'ecos-doclib__fileviewer_mobile': isMobile,
            'ecos-doclib__fileviewer_height-bc': isBreadcrumbsVisible && !isGroupActionsVisible,
            'ecos-doclib__fileviewer_height-ga': !isBreadcrumbsVisible && isGroupActionsVisible,
            'ecos-doclib__fileviewer_height-bc-ga': isBreadcrumbsVisible && isGroupActionsVisible
          })}
        >
          <FileList
            isMobile={isMobile}
            items={items}
            selected={selected}
            lastClicked={lastClicked}
            openFolder={openFolder}
            setSelected={setSelected}
            setLastClicked={setLastClicked}
            onDrop={onDrop}
          />
        </div>
      ) : (
        <Empty onDrop={onDrop} />
        /*t('document-library.empty-folder')*/
      );
  }

  return (
    <Well className="ecos-doclib__fileviewer-well">
      {isLoading && <Loader blur rounded />}
      {content}
    </Well>
  );
};

FilesViewer.propTypes = {
  stateId: PropTypes.string,
  isMobile: PropTypes.bool,
  fileViewer: PropTypes.shape({
    hasError: PropTypes.bool,
    isReady: PropTypes.bool,
    items: PropTypes.array
  }),
  openFolder: PropTypes.func,
  setSelected: PropTypes.func,
  groupActions: PropTypes.shape({
    isReady: PropTypes.bool,
    forRecords: PropTypes.shape({
      actions: PropTypes.array
    })
  }),
  path: PropTypes.array
};

export default FilesViewer;
