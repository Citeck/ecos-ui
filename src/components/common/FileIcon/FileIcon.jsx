import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import formats from './font/icons/high-contrast/catalog';
import './font/file-icon-high-contrast.css';
import './FileIcon.scss';

const FileIcon = ({ format, className }) => {
  let isCorrectFormat = false;
  let lowerCaseFormat = format || '';
  if (format && typeof format.toLowerCase === 'function') {
    lowerCaseFormat = format.toLowerCase();
    isCorrectFormat = Array.from(formats).includes(lowerCaseFormat);
  }

  return (
    <span
      className={classNames('ecos-file-icon', 'fiv-hct', className, {
        [`fiv-icon-${lowerCaseFormat}`]: isCorrectFormat,
        'fiv-icon-blank': !isCorrectFormat || lowerCaseFormat === 'blank'
      })}
    />
  );
};

FileIcon.propTypes = {
  className: PropTypes.string,
  format: PropTypes.oneOf(Array.from(formats))
};

export default FileIcon;
