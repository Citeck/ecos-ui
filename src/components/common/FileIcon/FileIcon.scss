.ecos-file-icon {
  &:before {
    content: '';
  }
}
