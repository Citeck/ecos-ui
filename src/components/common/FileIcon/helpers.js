import formats from './font/icons/high-contrast/catalog';
import { blankFormat } from './constants';

export function detectFormat(fileName) {
  try {
    if (typeof fileName !== 'string') {
      return blankFormat;
    }

    const parts = fileName.split('.');
    if (parts.length < 2) {
      return blankFormat;
    }

    const lastPart = parts[parts.length - 1].toLowerCase();
    if (Array.from(formats).includes(lastPart)) {
      return lastPart;
    }

    return blankFormat;
  } catch (e) {
    return blankFormat;
  }
}
