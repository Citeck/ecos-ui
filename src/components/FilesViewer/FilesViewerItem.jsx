import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import moment from 'moment';

import { ActionPropTypes } from '../common/grid/InlineTools/constants';
import { renderAction } from '../common/grid/InlineTools/helpers';
import EcosIcon from '../common/EcosIcon';
import FileIcon from '../common/FileIcon';
import { detectFormat } from '../common/FileIcon/helpers';
import { NODE_TYPES } from '../../constants/docLib';
import { useDropFile } from '../../hooks';

const DATE_FORMAT = 'DD.MM.YYYY HH:mm';

const FilesViewerItem = ({ item, isSelected, isLastClicked, isMobile, onClick, onDoubleClick, onDrop }) => {
  const { title, type, modified, actions, isEmpty = false } = item;
  const {
    flags: { isAboveDir },
    handlers
  } = useDropFile({ item, callback: onDrop });
  let extraProps = {};

  if (typeof onDrop === 'function') {
    extraProps = {
      ...extraProps,
      ...handlers
    };
  }

  const _onClick = e => typeof onClick === 'function' && onClick(item, e);
  const _onDoubleClick = e => typeof onDoubleClick === 'function' && onDoubleClick(item, e);

  let modifiedDisp = '-';

  if (modified) {
    const date = moment(modified);

    if (date.isValid()) {
      modifiedDisp = date.format(DATE_FORMAT);
    }
  }

  const actionsList = (actions || []).map((action, idx) => renderAction(action, idx));

  return (
    <div
      className={classNames('ecos-files-viewer__item', {
        'ecos-files-viewer__item_selected': isSelected || isAboveDir,
        'ecos-files-viewer__item_lastclicked': isLastClicked,
        'ecos-files-viewer__item_mobile': isMobile,
        'ecos-files-viewer__item_empty': isEmpty
      })}
      onClick={_onClick}
      onDoubleClick={_onDoubleClick}
      {...extraProps}
    >
      {!isEmpty && (
        <>
          <div className="ecos-files-viewer__item-left">
            <div className="ecos-files-viewer__item-icon-wrapper">
              {type === NODE_TYPES.DIR ? (
                <EcosIcon className={classNames('ecos-files-viewer__item-icon')} data={{ value: 'icon-folder' }} />
              ) : (
                <FileIcon className={classNames('ecos-files-viewer__item-file-icon')} format={detectFormat(title)} />
              )}
            </div>
            <span className="ecos-files-viewer__item-title" title={title}>
              {title}
            </span>
          </div>
          <div
            className={classNames('ecos-files-viewer__item-right', {
              'ecos-files-viewer__item-right_mobile': isMobile
            })}
          >
            <span className="ecos-files-viewer__item-modified" title={modifiedDisp}>
              {modifiedDisp}
            </span>
            <div className="ecos-files-viewer__item-actions">{actionsList}</div>
          </div>
        </>
      )}
    </div>
  );
};

const FilesViewerItemPropTypes = {
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  type: PropTypes.oneOf(Object.values(NODE_TYPES)),
  modified: PropTypes.string,
  actions: PropTypes.arrayOf(PropTypes.shape(ActionPropTypes))
};

FilesViewerItem.propTypes = {
  item: PropTypes.shape(FilesViewerItemPropTypes).isRequired,
  isSelected: PropTypes.bool,
  isLastClicked: PropTypes.bool,
  isMobile: PropTypes.bool,
  onClick: PropTypes.func,
  onDoubleClick: PropTypes.func,
  onDrop: PropTypes.func
};

export default FilesViewerItem;
export { FilesViewerItemPropTypes };
