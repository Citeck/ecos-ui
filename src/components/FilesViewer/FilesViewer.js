import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import uuidV4 from 'uuid/v4';

import { NODE_TYPES } from '../../constants/docLib';
import FilesViewerItem, { FilesViewerItemPropTypes } from './FilesViewerItem';
import './FilesViewer.scss';

const FilesViewer = ({ isDragged, isMobile, lastClicked, items: _items, selected, onClick, onDoubleClick, onDrop }) => {
  const hasTypeFile = _items.some(item => item.type === NODE_TYPES.FILE);
  const items = !hasTypeFile && isDragged ? [..._items, { isEmpty: true, id: uuidV4() }] : _items;

  return (
    <div className={classNames('ecos-files-viewer', { 'ecos-files-viewer_dragged-not-files': !hasTypeFile && isDragged })}>
      {items.map(item => (
        <FilesViewerItem
          key={item.id}
          item={item}
          isMobile={isMobile}
          isLastClicked={lastClicked === item.id}
          isSelected={selected.includes(item.id)}
          onClick={onClick}
          onDoubleClick={onDoubleClick}
          onDrop={onDrop}
        />
      ))}
    </div>
  );
};

FilesViewer.defaultProps = {
  items: [],
  selected: []
};

FilesViewer.propTypes = {
  isMobile: PropTypes.bool,
  isDragged: PropTypes.bool,
  items: PropTypes.arrayOf(PropTypes.shape(FilesViewerItemPropTypes)).isRequired,
  selected: PropTypes.arrayOf(PropTypes.string),
  lastClicked: PropTypes.string,
  onClick: PropTypes.func,
  onDoubleClick: PropTypes.func
};

export default FilesViewer;
