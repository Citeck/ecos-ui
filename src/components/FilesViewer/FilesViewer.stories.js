import React from 'react';

import { demoItems, demoActions } from './__mocks__/FilesViewer.mock';
import FilesViewer from './FilesViewer';

export default {
  title: 'Components/FilesViewer',
  component: FilesViewer,
  parameters: {
    actions: {
      argTypesRegex: '^on.*'
    }
  }
};

const defaultArgs = {
  items: demoItems
};

const Template = args => <FilesViewer {...args} />;

export const Default = Template.bind({});
Default.args = {
  ...defaultArgs
};

export const Selected = Template.bind({});
Selected.args = {
  ...defaultArgs,
  selected: [demoItems[1].id, demoItems[4].id, demoItems[5].id]
};

export const WithActions = Template.bind({});
WithActions.args = {
  ...defaultArgs,
  items: demoItems.map(item => ({ ...item, actions: [...demoActions] })),
  selected: [demoItems[1].id, demoItems[4].id]
};
