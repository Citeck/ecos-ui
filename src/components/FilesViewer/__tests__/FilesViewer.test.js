import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import React from 'react';

import FileIcon from '../../common/FileIcon';
import FilesViewer from '../FilesViewer';
import FilesViewerItem from '../FilesViewerItem';
import { file1, folder1, demoItems, demoActions } from '../__fixtures__/FilesViewer.fixtures';

describe('FilesViewer tests', () => {
  describe('<FilesViewer />', () => {
    it('should render FilesViewer component', () => {
      const { container } = render(<FilesViewer />);
      expect(container).toMatchSnapshot();
    });

    it('should render all items', () => {
      const { container } = render(<FilesViewer items={demoItems} />);
      expect(container.getElementsByClassName('ecos-files-viewer__item')).toHaveLength(demoItems.length);
    });

    it('should render selected items', () => {
      const selectedItems = [folder1.id, file1.id];
      const { container } = render(<FilesViewer items={demoItems} selected={selectedItems} />);
      expect(container.getElementsByClassName('ecos-files-viewer__item_selected')).toHaveLength(selectedItems.length);
    });
  });

  describe('<FilesViewerItem />', () => {
    it('should render FilesViewerItem component', () => {
      const { container } = render(<FilesViewerItem item={{ id: '', title: '' }} />);
      expect(container).toMatchSnapshot();
    });

    it('should render title', () => {
      const { container } = render(<FilesViewerItem item={file1} />);
      const title = container.getElementsByClassName('ecos-files-viewer__item-title');
      expect(title).toHaveLength(1);
      expect(title[0].textContent).toBe(file1.title);
    });

    describe('should render icon', () => {
      it('should render folder icon', () => {
        const { container } = render(<FilesViewerItem item={folder1} />);
        const icon = container.getElementsByClassName('icon-folder');
        expect(icon).toHaveLength(1);
      });

      it('should render file icon', () => {
        const { container } = render(<FilesViewerItem item={file1} />);
        expect(container.getElementsByClassName('ecos-file-icon')).toHaveLength(1);
      });
    });

    describe('should render modified', () => {
      const empty = '-';
      const emptyCases = [
        { input: undefined, output: empty },
        { input: null, output: empty },
        { input: '', output: empty }
      ];

      emptyCases.forEach((caseItem, idx) => {
        it(`case ${idx + 1}: "${caseItem.input}"`, () => {
          const item = {
            ...folder1,
            modified: caseItem.input
          };
          const { container } = render(<FilesViewerItem item={item} />);
          const modified = container.getElementsByClassName('ecos-files-viewer__item-modified');
          expect(modified[0].textContent).toBe(caseItem.output);
        });
      });

      it(`not empty case: "${folder1.modified}"`, () => {
        const { container } = render(<FilesViewerItem item={folder1} />);
        const modified = container.getElementsByClassName('ecos-files-viewer__item-modified');
        expect(modified[0].textContent).not.toBe(empty);
      });
    });

    it('should render selected', () => {
      const { container } = render(<FilesViewerItem item={file1} isSelected={true} />);
      expect(container.getElementsByClassName('ecos-files-viewer__item_selected')).toHaveLength(1);
    });

    it('should render actions', () => {
      const itemWithActions = {
        ...file1,
        actions: [...demoActions]
      };
      const { container } = render(<FilesViewerItem item={itemWithActions} isSelected={true} />);
      const actions = container.getElementsByClassName('ecos-files-viewer__item-actions')[0];
      expect(actions.children).toHaveLength(demoActions.length);
    });

    it(`should call 'onClick' when click by item`, async () => {
      const user = userEvent.setup();

      const onClick = jest.fn();
      const { container } = render(<FilesViewerItem item={file1} onClick={onClick} />);
      await user.click(container.getElementsByClassName('ecos-files-viewer__item')[0]);

      expect(onClick).toHaveBeenCalledTimes(1);
      expect(onClick.mock.calls[0][0]).toEqual(file1);
    });

    it(`should call 'onDoubleClick' when click by item twice`, async () => {
      const user = userEvent.setup();

      const onDoubleClick = jest.fn();
      const { container } = render(<FilesViewerItem item={file1} onDoubleClick={onDoubleClick} />);
      await user.dblClick(container.getElementsByClassName('ecos-files-viewer__item')[0]);

      expect(onDoubleClick).toHaveBeenCalledTimes(1);
      expect(onDoubleClick.mock.calls[0][0]).toEqual(file1);
    });
  });
});
