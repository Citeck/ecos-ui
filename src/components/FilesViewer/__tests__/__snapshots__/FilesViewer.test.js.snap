// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilesViewer tests <FilesViewer /> should render FilesViewer component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FilesViewer
    items={Array []}
    selected={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [],
      "className": "ecos-files-viewer",
    },
    "ref": null,
    "rendered": Array [],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [],
        "className": "ecos-files-viewer",
      },
      "ref": null,
      "rendered": Array [],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;

exports[`FilesViewer tests <FilesViewerItem /> should render FilesViewerItem component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FilesViewerItem
    item={
      Object {
        "id": "",
        "title": "",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="ecos-files-viewer__item-left"
        >
          <div
            className="ecos-files-viewer__item-icon-wrapper"
          >
            <UNDEFINED
              className="ecos-files-viewer__item-icon"
              data={
                Object {
                  "value": "icon-custom-file-empty",
                }
              }
            />
          </div>
          <span
            className="ecos-files-viewer__item-title"
            title=""
          >
            
          </span>
        </div>,
        <div
          className="ecos-files-viewer__item-right"
        >
          <span
            className="ecos-files-viewer__item-modified"
            title="-"
          >
            -
          </span>
          <div
            className="ecos-files-viewer__item-actions"
          />
        </div>,
      ],
      "className": "ecos-files-viewer__item",
      "onClick": [Function],
      "onDoubleClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="ecos-files-viewer__item-icon-wrapper"
            >
              <UNDEFINED
                className="ecos-files-viewer__item-icon"
                data={
                  Object {
                    "value": "icon-custom-file-empty",
                  }
                }
              />
            </div>,
            <span
              className="ecos-files-viewer__item-title"
              title=""
            >
              
            </span>,
          ],
          "className": "ecos-files-viewer__item-left",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <UNDEFINED
                className="ecos-files-viewer__item-icon"
                data={
                  Object {
                    "value": "icon-custom-file-empty",
                  }
                }
              />,
              "className": "ecos-files-viewer__item-icon-wrapper",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "className": "ecos-files-viewer__item-icon",
                "data": Object {
                  "value": "icon-custom-file-empty",
                },
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.memo),
                "compare": null,
                "type": [Function],
              },
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
              "className": "ecos-files-viewer__item-title",
              "title": "",
            },
            "ref": null,
            "rendered": "",
            "type": "span",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="ecos-files-viewer__item-modified"
              title="-"
            >
              -
            </span>,
            <div
              className="ecos-files-viewer__item-actions"
            />,
          ],
          "className": "ecos-files-viewer__item-right",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "-",
              "className": "ecos-files-viewer__item-modified",
              "title": "-",
            },
            "ref": null,
            "rendered": "-",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [],
              "className": "ecos-files-viewer__item-actions",
            },
            "ref": null,
            "rendered": Array [],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="ecos-files-viewer__item-left"
          >
            <div
              className="ecos-files-viewer__item-icon-wrapper"
            >
              <UNDEFINED
                className="ecos-files-viewer__item-icon"
                data={
                  Object {
                    "value": "icon-custom-file-empty",
                  }
                }
              />
            </div>
            <span
              className="ecos-files-viewer__item-title"
              title=""
            >
              
            </span>
          </div>,
          <div
            className="ecos-files-viewer__item-right"
          >
            <span
              className="ecos-files-viewer__item-modified"
              title="-"
            >
              -
            </span>
            <div
              className="ecos-files-viewer__item-actions"
            />
          </div>,
        ],
        "className": "ecos-files-viewer__item",
        "onClick": [Function],
        "onDoubleClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="ecos-files-viewer__item-icon-wrapper"
              >
                <UNDEFINED
                  className="ecos-files-viewer__item-icon"
                  data={
                    Object {
                      "value": "icon-custom-file-empty",
                    }
                  }
                />
              </div>,
              <span
                className="ecos-files-viewer__item-title"
                title=""
              >
                
              </span>,
            ],
            "className": "ecos-files-viewer__item-left",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <UNDEFINED
                  className="ecos-files-viewer__item-icon"
                  data={
                    Object {
                      "value": "icon-custom-file-empty",
                    }
                  }
                />,
                "className": "ecos-files-viewer__item-icon-wrapper",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "className": "ecos-files-viewer__item-icon",
                  "data": Object {
                    "value": "icon-custom-file-empty",
                  },
                },
                "ref": null,
                "rendered": null,
                "type": Object {
                  "$$typeof": Symbol(react.memo),
                  "compare": null,
                  "type": [Function],
                },
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
                "className": "ecos-files-viewer__item-title",
                "title": "",
              },
              "ref": null,
              "rendered": "",
              "type": "span",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="ecos-files-viewer__item-modified"
                title="-"
              >
                -
              </span>,
              <div
                className="ecos-files-viewer__item-actions"
              />,
            ],
            "className": "ecos-files-viewer__item-right",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "-",
                "className": "ecos-files-viewer__item-modified",
                "title": "-",
              },
              "ref": null,
              "rendered": "-",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [],
                "className": "ecos-files-viewer__item-actions",
              },
              "ref": null,
              "rendered": Array [],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
